tests/framework/portforward.go

```diff
@@ -0,0 +1,85 @@
+package framework
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	core "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetNGFPodName returns the name of the NGF Pod.
+func GetNGFPodName(
+	k8sClient client.Client,
+	namespace,
+	releaseName string,
+	timeout time.Duration,
+) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	var podList core.PodList
+	if err := k8sClient.List(ctx, &podList, client.InNamespace(namespace)); err != nil {
+		return "", fmt.Errorf("error getting list of Pods: %w", err)
+	}
+
+	for _, pod := range podList.Items {
```

Comment:
is it possible to pass a label selector as ListOptions in k8sClient.List ?

Reply:
Yep, good call.